src/receiver.ts

```diff
@@ -189,213 +248,390 @@ export class Receiver {
     })
   }
 
+  protected async callRequestHandler (requestId: number, amount: string, data: Buffer, keyId?: Buffer): Promise<{ fulfill: boolean, responseData: Buffer }> {
+    let fulfill = false
+    let finalized = false
+    let responseData = Buffer.alloc(0)
+
+    // This promise resolves when the user has either accepted or rejected the payment
+    await new Promise(async (resolve, reject) => {
+      // Reject the payment if:
+      // a) the user explicity calls reject
+      // b) if they don't call accept
+      // c) if there is an error thrown in the request handler
+      try {
+        await Promise.resolve(this.requestHandler({
```

Comment:
Isn't `await Promise.resolve(...` redundant? Why not a normal call?

Reply:
Promise.resolve can take a promise as a parameter, in which case it will just return that promise. So this code should allow the request handler to be synchronous or asynchonous